activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -740,6 +740,10 @@ def arel_visitor
 
         def build_statement_pool
         end
+
+        def build_result(columns, rows, column_types = {})
+          ActiveRecord::Result.new(columns, rows, column_types)
+        end
```

Comment:
> Extracting the construction of ActiveRecord::Result into a method like build_result will allow apps to hook into the method and add other connection-specific data to the Result instance.

Note that there's no guarantee of method signature stability with this change. It may be renamed, args tweaked, or refactored away anytime.

Recommend
* documenting its purpose as an internal, unofficial hook point in a comment
* passing through args/kwargs rather than giving a specific signature

Reply:
Good point @jeremy. I applied the changes in f735a2167c